packages/server/graphql/types/JiraServerIntegration.ts

```diff
@@ -33,6 +35,17 @@ const JiraServerIntegration = new GraphQLObjectType<{teamId: string; userId: str
   name: 'JiraServerIntegration',
   description: 'Jira Server integration data for a given team member',
   fields: () => ({
+    id: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'Composite key in jiraServer:providerId format',
+      resolve: async ({teamId, userId}: {teamId: string; userId: string}, _args, {dataLoader}) => {
+        const auth = await dataLoader
+          .get('teamMemberIntegrationAuths')
+          .load({service: 'jiraServer', teamId, userId})
+
+        return `jiraServer:${auth?.providerId}`
```

Comment:
If `auth` is undefined, we'll return `jiraServer:undefined` here. Could we check if `auth` exists, and return null if it doesn't?

Reply:
Also where is this `id` currently being used?

Reply:
@nickoferrall it is needed for relay to correctly update the store. We request `JiraServerintegration.searchQueries` after performing persist search query mutation.